layer/layer_store.go

```diff
@@ -294,7 +294,7 @@ func (ls *layerStore) registerWithDescriptor(ts io.Reader, parent ChainID, descr
 	if string(parent) != "" {
 		p = ls.get(parent)
 		if p == nil {
-			return nil, ErrLayerDoesNotExist
+			return nil, errors.WithStack(ErrLayerDoesNotExist)
```

Comment:
Orthogonal, but wondering if we should use `errdefs.NotFound()` for these; https://github.com/moby/moby/blob/7d4b7883815d45a0b4e1f4c50ec566e24040cb2a/errdefs/helpers.go#L13-L19

@cpuguy83

Reply:
I removed all the errors refactoring as it was not strictly required in this PR.